templates/protractor/page_objects/homepage/components/NavigationComponent.js

```diff
@@ -0,0 +1,26 @@
+import Wait from '../../../helpers/WaitHelper';
+import Timeouts from '../../../constants/Timeouts';
+import { selectors } from './NavigationComponent.po'
```

Comment:
1. I'm not a fan of the naming convention here: this class Is an actual page object. The imported selectors are just that: selectors :). Saying that, we do not have to slap `PageObject` to the name of every single one of them, let's just not overuse the term itself in places when it's not true.
2. What's the rationale for having them extracted to a separate file? Isn't jumping between two files less convenient?
3. Regardless of the above, there is still an inconsistency over here with some selectors exported and some inline.
4. Also, why not export whole `by` locators instead of only selectors?

Reply:
As far are selectors are concerned, they are moved to a separate file for the readability sake and keeping file as clean as possible- imagine you add 10, 15 elements you may need in the suite, not to mention if you add some selectors for elements visible only on mobile (then such selectors should be moved to another object, like `mobileSelector` or sth). Keeping it in one file would lead to littering file.
The inconsistency you're referring to is caused as in line 17 there's passed `linkName` variable to the selector. Surely, it would be possible to move partial selector to the `selectors` object, but still it would require some `inline` selectors.
Also, I do not believe that extracting `by` locators would an optimal solution, as in the current solution `NavigationComponent.po` file extracts simple JSON data, which may contain not only selectors object, but also e.x. modifiers object, like `{ isActive: 'is-active' }`